frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1058,6 +1096,33 @@ dbg:guard(ReaderHighlight, "onShowHighlightMenu",
             "onShowHighlightMenu must not be called with nil self.selected_text!")
     end)
 
+function ReaderHighlight:_getHighlightMenuAnchor()
+    local position = G_reader_settings:readSetting("highlight_dialog_position", "center")
+    if position == "center" then return end
+    local dialog_box = self.highlight_dialog:getContentSize()
+    local anchor_x = math.floor((self.screen_w - dialog_box.w) / 2) -- center by width
+    local screen_h = self.screen_h - Size.padding.small -- do not stick to the edge
+    local anchor_y, prefers_pop_down
+    if position == "top" then
+        anchor_y = Size.padding.small
+        prefers_pop_down = true
+    elseif position == "bottom" then
+        anchor_y = screen_h
```

Comment:
Lying about screen_h :) while you could probably do in the `bottom` branch like you do for the `top` branch: `anchor_y = screen_h - Size.padding.small`.
(I see that you reuse this hacked screen_h below - but the unbalance in this 2 branches reads odd. And no substraction computation saved, on the contrary it's avoided if "top", if you do it explicitely in the "bottom" and "gesture" branches.)